### ISSUE TITLE
embind does not support class methods qualified with "noexcept"

### ISSUE THREAD

**Issue by prideout** [emscripten-core/emscripten#11871]
This can be fixed by adding two new instantiations of `RegisterClassMethod` to `bind.h`, as seen here:

https://github.com/google/filament/pull/2940/commits/cddc82dceb7d968eec8de5f927027f7cfe0a9d8d

We should probably also update emscripten's unit tests to cover these cases.

To give some context, this will fix the following clang errors, which we ran into when upgrading the Filament build from C++14 to C++17:

```
/Users/prideout/github/emsdk/upstream/emscripten/system/include/emscripten/bind.h:1553:13: error: implicit instantiation of undefined template 'emscripten::internal::RegisterClassMethod<void (filament::View::*)(filament::View::AmbientOcclusion) noexcept>'
            invoker::template invoke<ClassType, Policies...>(methodName, callable);
            ^
../../web/filament-js/jsbindings.cpp:566:6: note: in instantiation of function template specialization 'emscripten::class_<filament::View, emscripten::internal::NoBaseClass>::function<emscripten::internal::DeduceArgumentsTag, void (filament::View::*)(filament::View::AmbientOcclusion) noexcept>' requested here
    .function("setAmbientOcclusion", &View::setAmbientOcclusion)
     ^
/Users/prideout/github/emsdk/upstream/emscripten/system/include/emscripten/bind.h:1285:16: note: template is declared here
        struct RegisterClassMethod;
               ^
```

**Comment by kripken** [comment]
cc @chadaustin @brion

**Comment by chadaustin** [comment]
@prideout I'd be in favor of such a pull request.

**Comment by david-fong** [comment]
I think this can be closed now that #17140 is merged.

**Comment by kripken** [comment]
Thanks @david-fong , closing.
